src/Appwrite/Messaging/Adapter/Realtime.php

```diff
@@ -311,11 +311,16 @@ public static function fromPayload(string $event, Document $payload, Document $p
                     }
 
                     $channels[] = 'documents';
-                    $channels[] = 'databases.' . $database->getId() .  '.collections.' . $payload->getAttribute('$collectionId') . '.documents';
-                    $channels[] = 'databases.' . $database->getId() . '.collections.' . $payload->getAttribute('$collectionId') . '.documents.' . $payload->getId();
-
+                    $channels[] = 'databases.' . $database->getId() .  '.collections.' . $payload[0]->getAttribute('$collectionId') . '.documents';
+                    if (count($payload) === 1) {
+                        $channels[] = 'databases.' . $database->getId() . '.collections.' . $payload[0]->getAttribute('$collectionId') . '.documents.' . $payload[0]->getId();
+                    }
+                    $payloadReads = [];
+                    foreach ($payload as $document) {
+                        $payloadReads = \array_merge($payloadReads, $document->getRead());
+                    }
```

Comment:
I dont' think we can do it like this 🤔 we need to make we only send document that a user can read, on that users connection. So here we probably need $roles to be a map of $documentId -> (collection + document permissions) then on the sending side, we need to check the permissions of each document against the user